mitiq/utils.py

```diff
@@ -385,3 +388,61 @@ def qem_methods() -> dict[str, str]:
     }
 
     return techniques
+
+
+def _count_gate_arities(circuit: cirq.Circuit) -> dict[str, int]:
+    """Counts gates in a circuit grouped by arity."""
+    counts = {"1q": 0, "2q": 0, "nq": 0}
+    for op in circuit.all_operations():
+        if isinstance(op.gate, MeasurementGate):
+            continue
+        n = len(op.qubits)
+        if n == 1:
+            counts["1q"] += 1
+        elif n == 2:
+            counts["2q"] += 1
+        else:
+            counts["nq"] += 1
+    return counts
+
+
+def compare_cost(
+    circuit: QPROGRAM,
+    qem_circuits: Sequence[QPROGRAM],
+    shots: int | None = None,
+) -> dict[str, int | dict[str, int]]:
+    """Compares the cost of a circuit to mitigated variants.
+
+    Args:
+        circuit: Original circuit before mitigation.
+        qem_circuits: Circuits generated by an error mitigation method.
+        shots: Optional number of shots per circuit.
+
+    Returns:
+        Dictionary summarizing circuit and gate overhead. If ``shots`` is
+        provided, the dictionary includes ``total_shots`` equal to ``shots``
+        multiplied by ``len(qem_circuits)``.
+    """
+
+    base = convert_to_mitiq(circuit)[0]
+    mitigated = [convert_to_mitiq(c)[0] for c in qem_circuits]
+
+    base_counts = _count_gate_arities(base)
+    total_counts = {"1q": 0, "2q": 0, "nq": 0}
+    for circ in mitigated:
+        counts = _count_gate_arities(circ)
+        for key in total_counts:
+            total_counts[key] += counts[key]
+
+    gate_overhead = {
+        key: total_counts[key] - base_counts.get(key, 0)
+        for key in total_counts
+    }
+
+    result: dict[str, int | dict[str, int]] = {
+        "extra_circuits": len(qem_circuits) - 1,
+        "gate_overhead": gate_overhead,
+    }
+    if shots is not None:
+        result["total_shots"] = shots * len(qem_circuits)
```

Comment:
Total shots should be kept constant when comparing experiments, so it makes more sense to divide total shots by the number of qem circuits.